t/ui/17-product-log.t

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+BEGIN {
+    unshift @INC, 'lib';
+}
+
+use Mojo::Base;
+use Test::More;
+use Test::Mojo;
+
+use OpenQA::Test::Case;
+use OpenQA::Client;
+
+use t::ui::PhantomTest;
+
+OpenQA::Test::Case->new->init_data;
+
+my $driver = t::ui::PhantomTest::call_phantom();
+if ($driver) {
+    plan tests => 22;
+}
+else {
+    plan skip_all => 'Install phantomjs and Selenium::Remote::Driver to run these tests';
+    exit(0);
+}
+
+my $t = Test::Mojo->new('OpenQA::WebAPI');
+# we need to talk to the phantom instance or else we're using wrong database
+my $url = 'http://localhost:' . t::ui::PhantomTest::get_mojoport;
+
+# Scheduled isos are only available to operators and admins
+$t->get_ok($url . '/admin/productlog')->status_is(302);
+$t->get_ok($url . '/login')->status_is(302);
+$t->get_ok($url . '/admin/productlog')->status_is(200);
+
+# Schedule iso - need UA change to add security headers
+# XXX: Test::Mojo loses it's app when setting a new ua
+# https://github.com/kraih/mojo/issues/598
+my $app = $t->app;
+$t->ua(OpenQA::Client->new(apikey => 'PERCIVALKEY02', apisecret => 'PERCIVALSECRET02')->ioloop(Mojo::IOLoop->singleton));
+$t->app($app);
+
+my $ret = $t->post_ok($url . '/api/v1/isos', form => {ISO => 'whatever.iso', DISTRI => 'opensuse', VERSION => '13.1', FLAVOR => 'DVD', ARCH => 'i586', BUILD => '0091'})->status_is(200);
+is($ret->tx->res->json->{count}, 10, '10 new jobs created');
```

Comment:
Test is needed, but I see that as not a problem. And you still can run this test standalone (as I usually do during development) so when something goes wrong you have early warning.